app/controllers/two_factor_authentication/otp_verification_controller.rb

```diff
@@ -127,6 +127,7 @@ def form_params
 
     def post_analytics(result)
       properties = result.to_h.merge(analytics_properties)
+      properties[:new_device] = current_user.new_device(cookies)
```

Comment:
more inlining
```suggestion
      properties = result.to_h.merge(analytics_properties, new_device: current_user.new_device(cookies))
```

Reply:
Implemented there and also on totp_verification_controller